### ISSUE TITLE
Tests for metrics

### ISSUE THREAD

**Issue by Vijayabhaskar96** [enochkan/torch-metrics#11]
Currently the folder tests doesn't have any tests, there should be tests for each metric implemented, to get the current answers for test cases we can use keras.metrics and sklearn mertics. Personally I have not used any testing libraries, keras used pytest in the past before merging with TF, should we follow the same?

**Comment by hassiahk** [comment]
Yes, even I am thinking the same and was looking at TF GitHub to see how they did it. I did not understand even a bit on how they have written test cases. I have zero experience in Unit Testing, maybe that's why.

**Comment by enochkan** [comment]
@hassiahk @Vijayabhaskar96 I can create some unit tests. Should not be too bad I've done it for my previous job, only in ReactJS tho haha

**Comment by Vijayabhaskar96** [comment]
I haven't written one but pytest looks easy and used by many popular frameworks, for example https://github.com/keras-team/keras-preprocessing uses it, and also they use Travis for CI, but I think we should use GitHub actions for CI.

**Comment by enochkan** [comment]
@Vijayabhaskar96 did you have experience with Travis/ GitHub actions? I'm not familiar with them ahha

**Comment by Vijayabhaskar96** [comment]
I don't have much experience with it either. We can figure out as we progress 😅, it should be simple enough we just have to write the yaml config files.

**Comment by Vijayabhaskar96** [comment]
I started writing tests in PyTest by referring PyTorch-lightning tests cases, once I'm clear with all inputs and outputs, I will set up Github actions for CI.

**Comment by Vijayabhaskar96** [comment]
So far I have written tests that compare results with sklearn metrics and if they match the test passes. This is the way even PyTorch-lightning does its tests. But some of the metrics don't exist in sklearn. For example, LogCosh. However it is available in keras metrics, can I use those? or there are any other metrics libraries? I personally don't like depending on TensorFlow as their API may change often and may become incompatible later. What's your opinion guys? @enochkan @hassiahk

**Comment by hassiahk** [comment]
We can use TensorFlow for now and change if the API changes in time or else we can define those metrics in numpy (this is provided in the docs of TensorFlow Losses API) and use those.

**Comment by Vijayabhaskar96** [comment]
I don't think reimplementing in numpy is a Good Idea, then we have the same metric implemented twice ourselves. Tensorflow is fine then?

**Comment by hassiahk** [comment]
Yes, we tested those with TensorFlow manually. So, we can use those. We will change the tests if any changes happen in TF API.
